### PR TITLE
Fix the comments added to the activity table

### PR DESCRIPTION
**Pull request by sharidas**
When user adds comments to a file, the activity
notifications were not updated. This change does
fix the issue.

Signed-off-by: Sujith Haridasan <sujith.h@gmail.com>

### CODE REVIEW

**Comment by sharidas** [comment]
### Summary
The comments added for files were missing in the activity app.

### Description
The comments added for the files were working in stable19 branch but were found to be not working in stable20 branch. This change helps to fix the issue.

### How was tested?
- Enable activity app
- Add a file to the Filesystem
- Add a comment to the file
- Navigate to the activity tab of the file, the comment is seen, just like stable19.

File: lib/Consumer.php [hunk 1]
```diff
@@ -68,11 +68,12 @@ public function receive(IEvent $event) {
 		$emailSetting = ($emailSetting) ? $this->userSettings->getUserSetting($event->getAffectedUser(), 'setting', 'batchtime') : false;
 
 		// Add activity to stream
-		if (!$selfAction) {
+		if ($selfAction) {
 			$activityId = $this->data->send($event);
 		}
```
Comment (nickvergessen) [inline lib/Consumer.php:73]:
```suggestion
		$activityId = $this->data->send($event);
```

File: lib/Consumer.php [hunk 2]
```diff
@@ -68,11 +68,12 @@ public function receive(IEvent $event) {
...
 
-		if (!$selfAction && $notificationSetting && $activityId) {
+		//Check if the activityId is set before using the variable directly in the condition.
+		if ($selfAction && $notificationSetting && isset($activityId)  && $activityId) {
```
Comment (nickvergessen) [inline lib/Consumer.php:76]:
```suggestion
		if (!$selfAction && $notificationSetting && $activityId) {
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]
This is how it should be.
Currently your own comments are missing, but with your current change the comments of others would be missing.

File: lib/Consumer.php (hunk 1 above)
Comment (sharidas) [reply lib/Consumer.php]: Done

File: lib/Consumer.php (hunk 2 above)
Comment (sharidas) [reply lib/Consumer.php]: Now I have verified with the scenario:

1. Created users `admin`, `user1`.
2. `admin` shares a file `foo.txt` to `user1`
3. `admin` comments on `foo.txt`. `admin` can see the comments and so does `user1`
4. `user1` comments on `foo.txt`. Both `admin` and `user1` can see the comments.

5. `admin` creates a file `bar.txt`
6. `admin` comments on the file`bar.txt` (self). The activity is available in the table.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**